types/reference/image_reference.go

```diff
@@ -0,0 +1,36 @@
+package reference
+
+import (
+	"fmt"
+
+	distreference "github.com/docker/distribution/reference"
+)
+
+// Parse parses the given references and return the repository and
+// tag (if present) from it. If there is an error during parsing, it will
+// return an error.
+func Parse(ref string) (string, string, error) {
+	distributionRef, err := distreference.ParseNamed(ref)
+	if err != nil {
+		return "", "", err
+	}
+
+	repository, tag := GetRepositoryAndTagFromNamedRef(distributionRef)
+	return repository, tag, nil
+}
+
+// GetRepositoryAndTagFromNamedRef returns a repository and tag given the specified reference.
+// This function is necessary as long as the docker "server" api make the distinction between repository
+// and tags.
+func GetRepositoryAndTagFromNamedRef(ref distreference.Named) (string, string) {
```

Comment:
I don't see the purpose of having this return both values, this could just be `GetTagFromNamedRef`, the `Repository` portion should just be got directly from `ref.Name()`.